src/chrome/content/scaffold/aceWrapper.js

```diff
@@ -37,6 +37,69 @@ window.addEventListener("DOMContentLoaded", function(e) {
 	JavaScriptMode = require("ace/mode/javascript").Mode;
 	TextMode = require("ace/mode/text").Mode;
 	EditSession = require("ace/edit_session").EditSession;
+
+	ace.config.loadModule('ace/ext/language_tools', function () {
+		var snippetManager = ace.require("ace/snippets").snippetManager;
+		snippets = [{
+			content: "for (let i=0; i<${1:arrayName}.length; i++) {\n\t${2:// continue here}\n}\n",
+			name: "simple for loop"
+		},
+		{
+			content: "var item = new Zotero.Item(${1:itemType});",
+			name: "new Zotero.Item"
+		},
+		{
+			content: "ZU.doGet(${1:url}, function(text) {\n\t${2:// continue here}\n});\n",
+			name: "ZU.doGet"
+		},
+		{
+			content: "ZU.doPost(${1:url}, ${2:body}, function(text) {\n\t${3:// continue here}\n});\n",
+			name: "ZU.doPost"
+		},
+		{
+			content: "ZU.processDocuments(${1:urls}, ${2:processor}, ${3:noCompleteOnError});\n",
+			name: "ZU.processDocuments"
+		},
+		{
+			content: "ZU.xpathText(${1:node}, '${2:xpath}');",
+			name: "ZU.xpathText"
+		},
+		{
+			content: "ZU.xpath(${1:node}, '${2:xpath}');",
+			name: "ZU.xpath"
+		}];
+		snippetManager.register(snippets);
+		
+		var ZoteroCompleter = {
+			getCompletions: function(editor, session, pos, prefix, callback) {
+				callback(null, [
+					{value: "ZU.cleanAuthor", score: 1000, meta: "Zotero Utility function"},
```

Comment:
Is this `meta` required? These seem a bit redundant. (If they need to stay, `Utility` should be lowercased.)

Reply:
Not sure, but it shows up in the suggestion selector, e.g.

![ace-autocompletion-meta](https://user-images.githubusercontent.com/5199995/36698145-5bc5293a-1b49-11e8-83ba-0af726a63572.png)